src/lib/flow/sol-flow-builder.c

```diff
@@ -973,8 +973,9 @@ sol_flow_builder_get_node_type(struct sol_flow_builder *builder)
     err = add_guards(builder);
     if (err < 0) {
         SOL_WRN("Failed to allocate memory for constructing node type");
+        free(opts);
         errno = ENOMEM;
-        return NULL;
+        goto guards_error;
```

Comment:
@edersondisouza I added the free(opts) line, then changed my mind to use the goto statement. So, it shouldn't be here. But, as @ceolin also submitted a patch to fix this, I dropped my pr.